test/e2e/service.go

```diff
@@ -1402,6 +1402,60 @@ var _ = framework.KubeDescribe("ESIPP [Slow][Feature:ExternalTrafficLocalOnly]",
 			framework.Failf("Source IP (%v) is not the client IP even after ESIPP turned on, expected a public IP.", clientIP)
 		}
 	})
+
+	It("should only allow access from service loadbalancer source ranges [Slow]", func() {
+		// this feature currently supported only on GCE/GKE/AWS
+		framework.SkipUnlessProviderIs("gce", "gke", "aws")
+
+		loadBalancerCreateTimeout := loadBalancerCreateTimeoutDefault
+		largeClusterMinNodesNumber := 100
+		if nodes := framework.GetReadySchedulableNodesOrDie(cs); len(nodes.Items) > largeClusterMinNodesNumber {
+			loadBalancerCreateTimeout = loadBalancerCreateTimeoutLarge
+		}
+
+		namespace := f.Namespace.Name
+		serviceName := "lb-sourcerange"
+		jig := NewServiceTestJig(cs, serviceName)
+
+		By("Prepare allow source ips")
+		// prepare the exec pods
+		// hostPod and acceptPod are allowed to access the loadbalancer
+		hostPod := framework.LaunchHostExecPod(cs, namespace, "hostexec-accept")
+		acceptPodName := createExecPodOrFail(cs, namespace, "execpod-accept")
+		dropPodName := createExecPodOrFail(cs, namespace, "execpod-drop")
+
+		hostPod, err := cs.Core().Pods(namespace).Get(hostPod.Name)
+		Expect(err).NotTo(HaveOccurred())
+		accpetPod, err := cs.Core().Pods(namespace).Get(acceptPodName)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("creating a pod to be part of the service " + serviceName)
+		// This container is an nginx container listening on port 80
+		// See kubernetes/contrib/ingress/echoheaders/nginx.conf for content of response
+		jig.RunOrFail(namespace, nil)
+		// Create loadbalancer service with source range from node[0] and podAccept
+		svc := jig.CreateTCPServiceOrFail(namespace, func(svc *api.Service) {
+			svc.Spec.Type = api.ServiceTypeLoadBalancer
+			svc.Spec.LoadBalancerSourceRanges = []string{hostPod.Status.HostIP + "/32", accpetPod.Status.PodIP + "/32"}
+		})
+
+		// Clean up loadbalancer service
+		defer func() {
+			jig.ChangeServiceType(svc.Namespace, svc.Name, api.ServiceTypeClusterIP, loadBalancerCreateTimeoutDefault)
```

Comment:
If we don't remove `LoadBalancerSourceRanges` filed from the service, it will fail on the validation since only LoadBalancer type service is allowed to have this filed set?